### ISSUE TITLE
thread 'main' panicked at 'function not loaded: clang_createIndex'

### ISSUE THREAD

**Issue by egrimley-arm** [KyleMayes/clang-sys#116]
This is more of a feature request than a bug, and it's basically the same issue as #115.

I did the following:
```
docker run -it debian:bullseye /bin/bash
apt-get update
apt-get install --no-install-recommends -y \
  ca-certificates cargo clang cmake git llvm make python3
cd
git clone https://github.com/parallaxsecond/parsec
cd parsec
git checkout 2a57277854e30344e4e6f3b55bd0866ffb3066b3
cargo build --features mbed-crypto-provider
```

And I got this error:
```
thread 'main' panicked at 'function not loaded: clang_createIndex', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/clang-sys-0.29.3/src/lib.rs:1659:1
```

It looks as though `parsec` requires `psa-crypto`, which requires `psa-crypto-sys`, which uses `bindgen`, which uses `clang-sys`, which panics because the library that contains `clang_createIndex` is not there.

The error does not occur if I `apt-get install libclang-dev`, so it was my fault for not installing that package.

However, I wonder if it would be possible to improve the error message, which makes it seem like an internal error in the tool chain and doesn't give someone who doesn't know much about the internals of `bindgen`, or perhaps doesn't even know what `bindgen` is, much of an idea of what they should do to fix the problem. Would it be possible for `clang-sys` to return an error code to `bindgen`, for example?

**Comment by KyleMayes** [comment]
`bindgen` will panic with a [different error message](https://github.com/rust-lang/rust-bindgen/blob/0996486f0977e12f1a64f7730b3eca81659aa839/src/lib.rs#L1896) if it fails to load a `libclang` shared library.

So I'm pretty sure you would only see this panic if `bindgen` was able to load a `libclang` shared library but that library did not have the `clang_createIndex` function. Which doesn't make much sense because I think this function has existed as long as `libclang` has.

In any case, the panic message can be improved to suggest the user install `libclang` and direct them to the appropriate documentation.

**Comment by KyleMayes** [comment]
Published as `v1.0.1`, created PR to update `bindgen`: https://github.com/rust-lang/rust-bindgen/pull/1897
